src/mixins/filing-template-mixin.ts

```diff
@@ -986,19 +986,19 @@ export default class FilingTemplateMixin extends DateMixin {
     const documentInfo = entitySnapshot.businessDocuments.documentsInfo
     this.setSpecialResolutionRules(
       {
-        name: filing.alteration?.rulesFileName || documentInfo.certifiedRules.name,
-        key: filing.alteration?.rulesFileKey || documentInfo.certifiedRules.key,
-        url: filing.alteration?.rulesFileKey ? null : entitySnapshot.businessDocuments.documents.certifiedRules,
+        name: filing.alteration?.rulesFileName || documentInfo?.certifiedRules?.name,
+        key: filing.alteration?.rulesFileKey || documentInfo?.certifiedRules?.key,
+        url: filing.alteration?.rulesFileKey ? null : entitySnapshot.businessDocuments.documents?.certifiedRules,
```

Comment:
Is this for the unit tests? (If so, please fix the tests instead of changing code.)

A warning about using this when setting to store: if the conditional chaining fails then it returns "undefined", which is not reactive. This can cause hard-to-find problems later. Unless you are sure your code handles undefined object values in the store then I recommend a fallback to null or empty string.

Reply:
No it's not for unit tests

Reply:
ok

Reply:
https://github.com/bcgov/lear/blob/476150e30a84c26c10129435a34c606ec3c10ca3/legal-api/src/legal_api/resources/v2/business/business_documents.py#LL125C46-L125C46

Reply:
https://github.com/bcgov/lear/blob/476150e30a84c26c10129435a34c606ec3c10ca3/legal-api/src/legal_api/resources/v2/business/business_documents.py#L108

Reply:
Let me double check this first

Reply:
I assume you're checking the reactivity of this store getter... please post what you find.

I know we've seen issues before where variables in the template didn't update because they weren't reactive (because they were accidentally set to undefined when resuming a draft filing). Dunno about the store... is it watching for object changes or object property changes? That's why I test a lot :)